features/query/timeout_due_to_no_active_workers/feature.py

```diff
@@ -34,8 +34,12 @@ async def check_result(runner: Runner, handle: WorkflowHandle):
     try:
         await handle.query(Workflow.simple_query, rpc_timeout=timedelta(seconds=1))
     except RPCError as e:
-        # Cancelled rather than deadline exceeded since the timeout is client-side
-        assert e.status == RPCStatusCode.CANCELLED
+        # Can be cancelled or deadline exceeded depending on whether client or
+        # server hit timeout first in a racy way
```

Comment:
Is this just a temporary work around?

Reply:
I cannot replicate deadline exceeded against a docker container or Temporalite that happened in #184. Given my inability to replicate, I have to assume that this is the server racing against core gRPC layer (tonic) to see which hits the deadline first.

I'd accept alternative suggestions here.